apply_factors_method.py

```diff
@@ -0,0 +1,49 @@
+
+import traceback
+import pandas as pd
+
+
+def _get_traceback(exception):
+    """
+    Given an exception, returns the traceback as a string.
+    :param exception: Exception object
+    :return: string
+    """
+
+    return ''.join(
+        traceback.format_exception(
+            etype=type(exception), value=exception, tb=exception.__traceback__
+        )
+    )
+
+
+def lambda_handler(event, context):
+    try:
+        working_dataframe = pd.DataFrame(event)
+
+        question_columns = ['Q601_asphalting_sand', 'Q602_building_soft_sand', 'Q603_concreting_sand', 'Q604_bituminous_gravel',
+                            'Q605_concreting_gravel', 'Q606_other_gravel', 'Q607_constructional_fill']
+
+        for question in question_columns:
+            # Loop through each question value, impute based on factor and previous value
+            # then drop the previous value and the imp factor
+            working_dataframe[question] = working_dataframe.apply(lambda x: x['prev_'+question]*x['imputation_factor_'+question], axis=1)
+            working_dataframe = working_dataframe.drop(['prev_'+question, 'imputation_factor_'+question], axis=1)
+
+    except Exception as exc:
```

Comment:
This exception block isn't being hit in the `test_apply_factors` test, your tests should cover sad-paths as well as happyt there also looks like there is test code in there e.g

- the `print` statement is a big code smell for me!
- doen't look like any of the import + purge code should be there.